kernel/src/process.rs

```diff
@@ -741,15 +751,19 @@ impl Process<'a> {
         (None, 0, 0)
     }
 
+    crate fn appid(&self) -> AppId {
+        AppId::new(self.kernel, self.app_idx)
+    }
+
     crate fn sbrk(&self, increment: isize) -> Result<*const u8, Error> {
         let new_break = unsafe { self.app_break.get().offset(increment) };
         self.brk(new_break)
     }
 
     crate fn brk(&self, new_break: *const u8) -> Result<*const u8, Error> {
-        if new_break < self.mem_start() || new_break >= self.mem_end() {
+        if new_break < self.allow_water_mark.get() || new_break >= self.mem_end() {
             Err(Error::AddressOutOfBounds)
-        } else if new_break > self.kernel_memory_break.get() {
+        } else if new_break > self.app_break.get() {
```

Comment:
I'm probably missing something, but wouldn't that mean that you would never be able to allocate more memory?

Reply:
Yes. Fixed and added comments inline to clarify whats going on (and to help me reason through it as I was fixing).